lib/src/repository/devices/devices_repository.dart

```diff
@@ -144,8 +142,22 @@ class DevicesRepository {
                     ?.value ??
                 false),
           );
+        } else if (trait.name
+            .toString()
+            .toLowerCase()
+            .contains('battery_level')) {
+          return BatteryLevelTrait(
+              'battery_level',
+              BatteryLevel(
+                  (trait as GgetDeviceData_device_traits__asBatteryLevelDeviceTrait)
+                          .state
+                          .percentage
+                          .reported
+                          ?.value ??
+                      0));
+        } else {
+          return UnknownTrait(trait.name.toString());
```

Comment:
This is getting out of control. Can we create a createMethod and pass name to it.